br/pkg/lightning/common/util_test.go

```diff
@@ -122,48 +121,48 @@ func (s *utilSuite) TestSQLWithRetry(c *C) {
+	require.Regexp(t, ".*context canceled", err.Error())
 
 	// query success
 	rows := sqlmock.NewRows([]string{"a"}).AddRow("1")
 	mock.ExpectQuery("select a from test.t1").WillReturnRows(rows)
 
 	err = sqlWithRetry.QueryRow(context.Background(), "", "select a from test.t1", aValue)
-	c.Assert(err, IsNil)
-	c.Assert(*aValue, Equals, 1)
+	require.NoError(t, err)
+	require.Equal(t, 1, *aValue)
 
 	// test Exec
 	mock.ExpectExec("delete from").WillReturnError(context.Canceled)
 	err = sqlWithRetry.Exec(context.Background(), "", "delete from test.t1 where id = ?", 2)
-	c.Assert(err, ErrorMatches, ".*context canceled")
+	require.Regexp(t, ".*context canceled", err.Error())
 
 	mock.ExpectExec("delete from").WillReturnResult(sqlmock.NewResult(0, 1))
 	err = sqlWithRetry.Exec(context.Background(), "", "delete from test.t1 where id = ?", 2)
-	c.Assert(err, IsNil)
+	require.NoError(t, err)
 
-	c.Assert(mock.ExpectationsWereMet(), IsNil)
+	require.Nil(t, mock.ExpectationsWereMet())
 }
 
-func (s *utilSuite) TestStringSliceEqual(c *C) {
-	c.Assert(common.StringSliceEqual(nil, nil), IsTrue)
-	c.Assert(common.StringSliceEqual(nil, []string{}), IsTrue)
-	c.Assert(common.StringSliceEqual(nil, []string{"a"}), IsFalse)
-	c.Assert(common.StringSliceEqual([]string{"a"}, nil), IsFalse)
-	c.Assert(common.StringSliceEqual([]string{"a"}, []string{"a"}), IsTrue)
-	c.Assert(common.StringSliceEqual([]string{"a"}, []string{"b"}), IsFalse)
-	c.Assert(common.StringSliceEqual([]string{"a", "b", "c"}, []string{"a", "b", "c"}), IsTrue)
-	c.Assert(common.StringSliceEqual([]string{"a"}, []string{"a", "b", "c"}), IsFalse)
-	c.Assert(common.StringSliceEqual([]string{"a", "b", "c"}, []string{"a", "b"}), IsFalse)
-	c.Assert(common.StringSliceEqual([]string{"a", "x", "y"}, []string{"a", "y", "x"}), IsFalse)
+func TestStringSliceEqual(t *testing.T) {
+	assert.True(t, common.StringSliceEqual(nil, nil))
+	assert.True(t, common.StringSliceEqual(nil, []string{}))
+	assert.False(t, common.StringSliceEqual(nil, []string{"a"}))
+	assert.False(t, common.StringSliceEqual([]string{"a"}, nil))
+	assert.True(t, common.StringSliceEqual([]string{"a"}, []string{"a"}))
+	assert.False(t, common.StringSliceEqual([]string{"a"}, []string{"b"}))
+	assert.True(t, common.StringSliceEqual([]string{"a", "b", "c"}, []string{"a", "b", "c"}))
+	assert.False(t, common.StringSliceEqual([]string{"a"}, []string{"a", "b", "c"}))
+	assert.False(t, common.StringSliceEqual([]string{"a", "b", "c"}, []string{"a", "b"}))
+	assert.False(t, common.StringSliceEqual([]string{"a", "x", "y"}, []string{"a", "y", "x"}))
 }
 
-func (s *utilSuite) TestInterpolateMySQLString(c *C) {
-	c.Assert(common.InterpolateMySQLString("123"), Equals, "'123'")
-	c.Assert(common.InterpolateMySQLString("1'23"), Equals, "'1''23'")
-	c.Assert(common.InterpolateMySQLString("1'2''3"), Equals, "'1''2''''3'")
+func TestInterpolateMySQLString(t *testing.T) {
+	assert.Equal(t, "'123'", common.InterpolateMySQLString("123"))
```

Comment:
Please use ```require``` instead of ```assert```.

Reply:
I think `assert` is suitable here. The 3 lines are totally independent. So we need to continue test other cases when one of them is failed.

Reply:
It can work for such cases so I'll approve this. However, I tend to the comment in the other PR that the benefit doesn't pay for using two flavor and confuses others which one to use. Almost, where using `assert` can use `require`, but where using `require` cannot always use `assert`.

Reply:
> confuses others

This means whether we push every contributors write test should learn the difference between these two flavors.